### PR TITLE
Add footer feature.

### PR DESCRIPTION
**Pull request by pedrovgs**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,14 +73,14 @@ public class DragonBallAdapter extends HeaderRecyclerViewAdapter<RecyclerView.Vi
...
+  DragonBallFooter footer = getFooter();
   adapter.setHeader(header);
+  adapter.setFooter(footer);
   adapter.setItems(characters);
   recyclerView.setAdapter(adapter);
 
 ```
 
-**You can use methods like ``isHeaderType(viewType)`` or ``isHeaderPosition(position)`` to return the ViewHolder associated to the header or to draw the view associated to the header. The header instance can be obtained using the method ``getHeader()``**.
-
```
Comment (pedrovgs) [inline README.md:70]: This is not so important xD

File: library/src/test/java/com/karumi/headerrecyclerview/HeaderSpanSizeLookupTest.java [hunk 2]
```diff
@@ -69,7 +79,7 @@ public void shouldReturnHeaderSpanSizeIfTheAdapterHasNoConfiguredAHeaderAndThePo
     assertEquals(1, headerSpanSizeLookup.getSpanSize(POSITION_ONE));
   }
 
-  private Object giveAHeader() {
+  private Object givenAHeader() {
```
Comment (pedrovgs) [inline library/src/test/java/com/karumi/headerrecyclerview/HeaderSpanSizeLookupTest.java:82]: There was a typo here.

**Comment by pedrovgs** [comment]
@flipper83 can you take a look here?

File: library/src/main/java/com/karumi/headerrecyclerview/HeaderRecyclerViewAdapter.java [hunk 3]
```diff
@@ -107,14 +132,30 @@ public void setItems(List<T> items) {
...
+  public void setFooter(F footer) {
+    this.footer = footer;
+  }
+
+  public void showFooter() {
+    this.showFooter = true;
+  }
+
+  public void hideFooter() {
+    this.showFooter = false;
```
Comment (flipper83) [inline library/src/main/java/com/karumi/headerrecyclerview/HeaderRecyclerViewAdapter.java:145]: yo don't need to do a notifydatasetchange?

**Comment by flipper83** [comment]
:+1:

**Pull request merged**